airflow/models/taskinstance.py

```diff
@@ -317,11 +317,14 @@ def clear_task_instances(
         )
         dag_run_state = DagRunState(dag_run_state)  # Validate the state value.
         for dr in drs:
-            dr.state = dag_run_state
-            dr.start_date = timezone.utcnow()
-            if dag_run_state == DagRunState.QUEUED:
-                dr.last_scheduling_decision = None
-                dr.start_date = None
+            # There is no need to reset anything in the middle of a running DagRun.
+            # Only reset things for already finished runs.
```

Comment:
This needs to be a part of the docstring to describe what the function does. And I don’t think it’s good enough to just say “there is no need”—why is there no need?